Assets/Scripts/Characters/Humans/Hero.cs

```diff
@@ -3207,56 +3203,51 @@ public void Launch(Vector3 des, bool left = true, bool leviMode = false)
+                Bullet.Source.BeltLeft => hookRefL1,
+                Bullet.Source.BeltRight => hookRefR1,
+                Bullet.Source.GunLeft => hookRefL2,
+                Bullet.Source.GunRight => hookRefR2,
+                _ => throw new ArgumentOutOfRangeException(nameof(source), source, null)
+            };
 
-                GameObject obj2 = !useGun ? hookRefL1 : hookRefL2;
-                string str = !useGun ? "hookRefL1" : "hookRefL2";
-                hookLeft.transform.position = obj2.transform.position;
-                float num = !single ? ((distance <= 50f) ? (distance * 0.05f) : (distance * 0.3f)) : 0f;
-                Vector3 vector = (point - ((transform.right * num))) - hookLeft.transform.position;
-                vector.Normalize();
-                if (mode == 1)
-                {
-                    hookLeft.launch((vector * 3f), Rigidbody.velocity, str, true, gameObject, true);
-                }
-                else
-                {
-                    hookLeft.launch((vector * 3f), Rigidbody.velocity, str, true, gameObject, false);
-                }
+            var startPos = transform.position;
+            var startRot = transform.rotation;
+            var hook = PhotonNetwork.Instantiate("hook", startPos, startRot, 0).GetComponent<Bullet>();
+            if (Bullet.IsLeft(source))
+            {
+                hookLeft = hook;
+                LaunchHook(hook);
                 launchPointLeft = Vector3.zero;
             }
-        }
-
-        public void LaunchRightRope(float distance, Vector3 point, bool single, int mode = 0)
-        {
-            if (currentGas != 0f)
+            else
             {
-                UseGas(0f);
-                hookRight = PhotonNetwork.Instantiate("hook", transform.position, transform.rotation, 0).GetComponent<Bullet>();
-
-                GameObject obj2 = !useGun ? hookRefR1 : hookRefR2;
-                string str = !useGun ? "hookRefR1" : "hookRefR2";
-                hookRight.transform.position = obj2.transform.position;
-                float num = !single ? ((distance <= 50f) ? (distance * 0.05f) : (distance * 0.3f)) : 0f;
-                Vector3 vector = (point + ((transform.right * num))) - hookRight.transform.position;
-                vector.Normalize();
-                if (mode == 1)
-                {
-                    hookRight.launch((vector * 5f), Rigidbody.velocity, str, false, gameObject, true);
-                }
-                else
-                {
-                    hookRight.launch((vector * 3f), Rigidbody.velocity, str, false, gameObject, false);
-                }
+                hookRight = hook;
+                LaunchHook(hook);
                 launchPointRight = Vector3.zero;
             }
+
+            void LaunchHook(Bullet hook)
+            {
+                var num = !single ? distance <= 50f ? distance * 0.05f : distance * 0.3f : 0f;
```

Comment:
Code smell: "Extract this nested ternary operation into an independent statement."